src/main/java/org/devnull/jedi/PowerDNSConnectionHandler.java

```diff
@@ -125,7 +126,14 @@ public void run()
 					log.debug("waiting to read request from socket");
 				}
 
-				requestLine = reader.readLine();
+				try {
+					requestLine = reader.readLine();
+				} catch (IOException e){
+					//expected, PowerDNS doesn't tell us when it's done using a connection, we need a timeout to clean up correctly.
+					log.debug("socket timed out, closing");
```

Comment:
We're not closing the socket here, but we can log the timeout.  We should also emit a statsd counter for it.